Client/Frontend/Browser/Playlist/Utilities/PlaylistMediaStreamer.swift

```diff
@@ -0,0 +1,189 @@
+            MPMediaItemPropertyPlaybackDuration: item.duration,
+            MPNowPlayingInfoPropertyPlaybackProgress: 0.0,
+//            MPNowPlayingInfoPropertyDefaultPlaybackRate: 1.0,
+            MPNowPlayingInfoPropertyAssetURL: URL(string: item.pageSrc) as Any,
+            MPNowPlayingInfoPropertyElapsedPlaybackTime: 0.0, //player.currentTime.seconds
+            //MPNowPlayingInfoPropertyPlaybackQueueIndex: 0,
+            //MPNowPlayingInfoPropertyPlaybackQueueCount: 0
+        ]
+    }
+    
+    static func clearNowPlayingInfo() {
+        MPNowPlayingInfoCenter.default().nowPlayingInfo = nil
+    }
+    
+    static func setNowPlayingMediaArtwork(image: UIImage?) {
+        if let image = image {
+            let artwork = MPMediaItemArtwork(boundsSize: image.size, requestHandler: { _ -> UIImage in
+                // Do not resize image here.
+                // According to Apple it isn't necessary to use expensive resize operations
+                return image
+            })
+            setNowPlayingMediaArtwork(artwork: artwork)
+        }
+    }
+    
+    static func setNowPlayingMediaArtwork(artwork: MPMediaItemArtwork?) {
+        if let artwork = artwork {
+            MPNowPlayingInfoCenter.default().nowPlayingInfo?[MPMediaItemPropertyArtwork] = artwork
+        }
+    }
+    
+    static func getMimeType(_ url: URL, _ completion: @escaping (String?) -> Void) {
+        let request: URLRequest = {
+            var request = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalCacheData, timeoutInterval: 10.0)
+            
+            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range
+            request.addValue("bytes=0-1", forHTTPHeaderField: "Range")
+            request.addValue(UUID().uuidString, forHTTPHeaderField: "X-Playback-Session-Id")
+            request.addValue(UserAgent.shouldUseDesktopMode ? UserAgent.desktop : UserAgent.mobile, forHTTPHeaderField: "User-Agent")
+            return request
+        }()
+        
+        URLSession(configuration: .ephemeral).dataTask(with: request) { data, response, error in
+            DispatchQueue.main.async {
+                if let error = error {
+                    log.error("Error fetching MimeType for playlist item: \(url) - \(error)")
+                    return completion(nil)
+                }
+                
+                if let response = response as? HTTPURLResponse, response.statusCode == 302 || response.statusCode >= 200 && response.statusCode <= 299 {
+                    if let contentType = response.allHeaderFields["Content-Type"] as? String {
+                        completion(contentType)
+                        return
+                    } else {
+                        completion("video/*")
+                        return
+                    }
+                }
+                
+                completion(nil)
+            }
+        }.resume()
```

Comment:
Don't we need `finishTasksAndInvalidate` after resume here?
I remember we had to add it to my BraveSearch Combine code, not sure if we need it in non-Combine places too